week05/src/TodoItem.js

```diff
@@ -0,0 +1,23 @@
+function TodoItem({ todoThis, todoList, setTodoList }) {
+    function deleteTodo() {
+        setTodoList(todoList.filter(todoItem => todoItem.id !== todoThis.id));
+    }
+
+    function completeTodo() {
+        setTodoList(
+            todoList.map((todoItem) => {
+                return todoItem.id === todoThis.id ? { ...todoThis, done: !todoThis.done } : todoItem;
+            })
+        );
+    }
+
+    return(
+        <li className="todo-item">
+            <input type="checkbox" checked={todoThis.done} onClick={completeTodo} className="checkbox" />
```

Comment:
P4: input 태그에서 checked를 사용할 경우, onChange 핸들러를 사용하지 않으면 warning이 뜹니다. onClick을 사용하여 변경해도 좋지만, 만약 경고가 뜨는 것이 싫다면 onClick 대신 onChange를 사용해도 됩니다.
```suggestion
        <li className="todo-item">
            <input type="checkbox" checked={todoThis.done} onChange={completeTodo} className="checkbox" />
```